README.md

```diff
@@ -186,6 +186,32 @@ getFID(sendToGoogleAnalytics);
 getLCP(sendToGoogleAnalytics);
 ```
 
+### Send the results to Google Tag Manager
+
+The following example measures each of the Core Web Vitals metrics and sends them as seperate dataLayer-events to be used by Google Tag Manager. With the `web-vitals` trigger you send the metrics to any tag inside your account.
+
+```js
+import {getCLS, getFID, getLCP} from 'web-vitals';
+
+function sendToGoogleAnalytics({name, delta, id}) {
+  // Assumes the global `dataLayer` array exists, see:
+  // https://developers.google.com/tag-manager/devguide
+  dataLayer.push({
+      event: 'web-vitals',
```

Comment:
Do you know what this value is used for? Is it sent to Google Analytics or just used internally by GTM?

Reply:
This key is what you build a trigger on in GTM. The value just have to match whatever you set in the trigger on the GTM side.

Reply:
Not sent anywhere unsless you want to.

Reply:
I see, so then the you'd set up a trigger in your analytics tag in GTM to listen for the `web-vitals` event, and then configure how you want to send the data to Google Analytics?

Do the `event_*` properties automatically map, or would a user have to set that up themselves? (Sorry, never used GTM).

Reply:
Let me give you a quick GTM workflow for setting this up:

1. **Create a GA Event Tag**
![chrome_2020-05-10_23-01-28](https://user-images.githubusercontent.com/3629563/81510510-76cc0d00-9312-11ea-88da-80abd59adbd8.png)

2. **Map Event Tracking Parameters**
Here you need to manually map each "dataLayer"-variable seperate. So there is actually no automatic mapping here.
![bilde](https://user-images.githubusercontent.com/3629563/81510532-aa0e9c00-9312-11ea-92a7-66c3b81c53cc.png)

3. **Define and set a trigger**
Here GTM reads from the `event` key defined at the top of the dataLayer.push()-object.
![chrome_2020-05-10_23-07-22](https://user-images.githubusercontent.com/3629563/81510588-1e493f80-9313-11ea-8988-e97df4024833.png)

4. **Pull it all together**
![chrome_2020-05-10_23-08-39](https://user-images.githubusercontent.com/3629563/81510605-3a4ce100-9313-11ea-87e8-ef0edd213fc2.png)

And it will look something like this. Every time a new dataLayer.push() is run, it will send an event to Google Analytics. 

You can of course set more complex triggers, say if you only want to send the events on certain pages of your website or any other criteria that you might think of. I'm thinking of running this only on product-pages for a ecommerce client of mine as a start.

Reply:
Gotcha, thanks for clarifying. Do you know of any good documentation that we could link to for someone who's a GTM user but might not be familiar with how to set this up?

Reply:
The official documentation is rather lacking in the particualar area. 
I linked the dev-doc in the Readme for the `dataLayer` part of the story. 

The configuration part you have [the offical Tag Manager docs](https://support.google.com/tagmanager/answer/6106716?hl=en&ref_topic=6333310l). But its probably better to link to something from [analyticsmania.com](https://www.analyticsmania.com/post/google-tag-manager-event-tracking/#section3)

Reply:
Thanks, I'll merge this and then figure how where to link to in a separate PR.

Reply:
I could probably write up the walk-through comment as a gist or something, for a simple step-by-step guide, and we could link to in the docs.

Reply:
Sounds good to me. Longer term I'd like to expand that docs to individual guides that show how to do all of these types of things. And I'd also like to open it up to contributions from other analytics providers.

If you write up a gist that'd be a good starting point for the time being.

Reply:
Hello, I had to look up this thread to figure out how to set up GTM. Why not create a new file and slap this comment in there: https://github.com/GoogleChrome/web-vitals/pull/28#discussion_r422701126

Reply:
Added in https://github.com/GoogleChrome/web-vitals/commit/1aa8f7a198aba63039b9c579e0a928c5dee68f23

Reply:
Thanks for this, Simen. In addition to sending it to GA as Event, FID and LCP can be sent as Timing events to GA via GTM.

Reply:
@madilk I'd recommend not sending these to GA as timing hits, as those are quite restricted (e.g. are sampled at 1%, can't be used in segments, etc.)

Reply:
@philipwalton Perhaps, it might be better to send as Events and Timing hits to give the flexibility later? Usually, we set the siteSpeedSampleRate field in GA hits to be much higher % than the default ~1%. Also, I wasn't sure about the timing segment and checked it out. Timing hit values can be used in custom segments within GA. Attaching a screenshot here.
![timing](https://user-images.githubusercontent.com/20192415/82672033-d69ab000-9c50-11ea-8dcb-8aae6b4beba3.JPG)

Reply:
Interesting, maybe the sampling rules for timing hits has changed since I last tried them, but regardless the 1% sampling is the biggest blocker.

Reply:
And to clarify, the 1% sampling rate is applied on both the client and the server, so changing the 1% to 100% will not actually get you 100%. See: https://developers.google.com/analytics/devguides/collection/analyticsjs/user-timings#sampling_considerations

Reply:
hello @SimenHansen,
thanks a lot for the GTM workflow, it is very useful!
now that i can track the events in analytics, i'd like to go one step further, and get a quick visual repartition of "good"/"ok"/"poor" performances..
by any chance, would you have any idea of how this could be achieved using data studio?
let's say to produce graphs like the ones in CrUX Dashboard:
![image](https://user-images.githubusercontent.com/6266567/99531148-99ff0580-29a2-11eb-992e-650165123480.png)

because having avg. values in analytics is better than nothing but does not help to get the full picture